internal/runbits/runtime/request.go

```diff
@@ -0,0 +1,59 @@
+package runtime
+
+import (
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+type Request struct {
+	auth           *authentication.Auth
+	analytics      analytics.Dispatcher
+	project        *project.Project
+	namespace      *model.Namespace
+	customCommitID *strfmt.UUID
+	trigger        target.Trigger
+	svcModel       *model.SvcModel
+	config         Configurable
+	opts           Opts
+	asyncRuntime   bool
+}
+
+func NewRequest(auth *authentication.Auth,
+	an analytics.Dispatcher,
+	proj *project.Project,
+	customCommitID *strfmt.UUID,
+	trigger target.Trigger,
+	svcm *model.SvcModel,
+	cfg Configurable,
+	opts Opts,
+) *Request {
+
+	return &Request{
```

Comment:
The idea with this type was to have a central mechanism for controlling and manipulating how the runtime sourcing behaves. This may be overkill for the use-case of this story and we could do this check in the actual refresh functions.